src/worker.ts

```diff
@@ -103,21 +121,35 @@ async function getDataFeedSlice(
   } else {
     debug('getDataFeedSlice already cached: %s', sliceKey)
   }
-  // everything went well (already cached or successfull cached) let's communicate it to parent thread
-  const message: WorkerMessage = {
-    sliceKey,
-    slicePath
-  }
-  parentPort!.postMessage(message)
+
+  return slicePath
 }
 
+function getDownloadOptionsFromFilters(_cacheDir: string, exchange: string, filters: Filter<any>[]) {
+  return filters.reduce((acc, filter) => {
+    if (!filter.symbols) {
+      const cacheDir = `${_cacheDir}/feeds/${exchange}/__all__/${filter.channel}`
+      acc.push({ cacheDir, filters: [{ channel: filter.channel }] })
+    } else {
+      filter.symbols.forEach((symbol) => {
+        const cacheDir = `${_cacheDir}/feeds/${exchange}/${symbol}/${filter.channel}`
+        acc.push({ cacheDir, filters: [{ channel: filter.channel, symbols: [symbol] }] })
+      })
+    }
+    return acc
+  }, [] as DownloadOption[])
+}
+
+type DownloadOption = { cacheDir: string; filters: Filter<any>[] }
+
```

Comment:
with `fineGrainCache` each `filter` will correspond to one `cacheDir`. So we need to pass multiple `{ cacheDir, filters } as DownloadOption` to `getDataFeedSlice()`.